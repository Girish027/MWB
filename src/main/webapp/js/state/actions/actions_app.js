
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
