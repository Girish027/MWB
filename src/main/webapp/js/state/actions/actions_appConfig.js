import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
