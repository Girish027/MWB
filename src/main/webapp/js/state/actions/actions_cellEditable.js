import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
