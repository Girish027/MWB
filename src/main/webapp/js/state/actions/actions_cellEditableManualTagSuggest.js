import store from 'state/configureStore';
import errorMessageUtil from 'utils/ErrorMessageUtil';
import getUrl, { pathKey } from 'utils/apiUrls';
import * as types from './types';

let controller;
export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_CREATE,
  stateKey,
});
export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_REMOVE,
  stateKey,
});

const getSuggestedTagsRequest = ({ stateKey, projectId, intent }) => ({
  type: types.GET_SUGGESTED_TAGS_REQUEST,
  stateKey,
  projectId,
  intent,
});

const getSuggestedTagsError = ({
  stateKey, projectId, intent, error,
}) => ({
  type: types.GET_SUGGESTED_TAGS_ERROR,
  stateKey,
  projectId,
  intent,
  error,
});

const getSuggestedTagsSuccess = ({
  stateKey, projectId, intent, suggestedTags,
}) => ({
  type: types.GET_SUGGESTED_TAGS_SUCCESS,
  stateKey,
  projectId,
  intent,
  suggestedTags,
});

export const getSuggestedTags = ({ stateKey, projectId, intent }) => (dispatch, getState) => new Promise((resolve, reject) => {
  if (controller) {
    controller.abort();
  }
  controller = new AbortController();
  const signal = controller.signal;
  dispatch(getSuggestedTagsRequest({ stateKey, projectId, intent }));
  const state = getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;

  const fetchUrl = getUrl(pathKey.suggestedTags, { projectId, intent, clientId });

  fetch(fetchUrl, {
    signal,
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      const suggestedTags = json && json.length ? json : [];
      const intentIndex = suggestedTags.indexOf(intent);
      if (intentIndex >= 0) {
        suggestedTags.slice(intentIndex, 1);
      }
      dispatch(getSuggestedTagsSuccess({
        stateKey, projectId, intent, suggestedTags,
      }));
      resolve([...suggestedTags]);
    })
    .catch((error) => {
      if (error.name === 'AbortError') { return; }
      dispatch(getSuggestedTagsError({
        stateKey, projectId, intent, error,
      }));
      reject(error);
    });
});

export const cursorDown = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_DOWN,
  stateKey,
});

export const cursorUp = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_UP,
  stateKey,
});
