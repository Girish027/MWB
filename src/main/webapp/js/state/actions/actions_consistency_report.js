import errorMessageUtil from 'utils/ErrorMessageUtil';
import store from 'state/configureStore';

import getUrl, { pathKey } from 'utils/apiUrls';
import * as appActions from './actions_app';
import * as types from './types';

export const reset = () => ({
  type: types.CONSISTENCY_REPORT_RESET,
});

export const setProjectId = projectId => ({
  type: types.CONSISTENCY_REPORT_SET_PROJECT_ID,
  projectId,
});

export const setProject = ({ projectId, project }) => ({
  type: types.CONSISTENCY_REPORT_SET_PROJECT,
  projectId,
  project,
});

export const setDatasets = ({ projectId, datasets }) => ({
  type: types.CONSISTENCY_REPORT_SET_DATASETS,
  projectId,
  datasets,
});


export const setFilter = ({ datasets, onlyConflicts }) => ({
  type: types.CONSISTENCY_REPORT_SET_FILTER,
  datasets,
  onlyConflicts,
});

const requestSearchSuccess = ({ results, startIndex }) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_SUCCESS,
  results,
  startIndex,
});

const requestSearchRequest = ({
  promise, projectId, query, filter, sort, startIndex, limit,
}) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_REQUEST,
  promise,
  projectId,
  query,
  filter,
  sort,
  startIndex,
  limit,
});

const requestSearchError = ({ error }) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_ERROR,
  error,
});

export const setPagerSettings = ({ startIndex, limit, showControls }) => ({
  type: types.CONSISTENCY_REPORT_SET_PAGER_SETTINGS,
  startIndex,
  limit,
  showControls,
});

export const requestSearch = ({
  projectId, query, filter, sort, startIndex, limit, delay,
}, isNested) => (dispatch, getState) => {
  const promise = new Promise((resolve, reject) => {
    const state = getState();
    const { userId, csrfToken } = state.app;
    const clientId = state.header.client.id;

    const fetchUrl = getUrl(pathKey.consistency, {
      projectId,
      startIndex,
      limit,
      sortProperty: sort.property,
      sortDirection: sort.direction,
      clientId,
    });

    const filterBodyObj = {};
    filterBodyObj.onlyTagged = true;
    filterBodyObj.onlyConflicts = filter.onlyConflicts;
    if (filter.datasets && filter.datasets.length) {
      filterBodyObj.datasetIds = filter.datasets;
    }

    setTimeout(() => {
      fetch(fetchUrl, {
        credentials: 'same-origin',
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'X-CSRF-TOKEN': csrfToken,
        },
        body: JSON.stringify({
          configId: '0',
          filter: filterBodyObj,
        }),
      })
        .then(errorMessageUtil.handleErrors)
        .then(response => response.json())
        .then((results) => {
          const searchResults = results.transcriptionList || [];

          searchResults.forEach((res, index) => {
            res._rowIndex = index;
          });

          if (!searchResults.length && startIndex > 0) {
            const total = results.total;
            const maxPage = Math.max(Math.floor(total / limit) + (total % limit ? 1 : 0), 1);
            const newStartIndex = (maxPage - 1) * limit;
            dispatch(requestSearch({
              projectId,
              query,
              filter,
              sort,
              startIndex: newStartIndex,
              limit,
            }, true))
              .then((results) => {
                dispatch(requestSearchSuccess({ results, startIndex: newStartIndex }));
                resolve(results);
              })
              .catch((error) => {
                reject(error);
              });
          } else {
            if (!isNested) {
              dispatch(requestSearchSuccess({ results, startIndex }));
            }
            resolve(results);
          }
        })
        .catch((error) => {
          dispatch(requestSearchError({ error }));
          errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
          reject(error);
        });
    }, delay || 0);
  });

  if (!isNested) {
    dispatch(requestSearchRequest({
      promise, projectId, query, filter, sort, startIndex, limit,
    }));
  }

  return promise;
};

export const refresh = ({ delay }) => ({
  type: types.CONSISTENCY_REPORT_REFRESH,
  delay,
});
