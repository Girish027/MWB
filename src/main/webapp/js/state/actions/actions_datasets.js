
import fetch from 'isomorphic-fetch';
import queryString from 'query-string';
import store from 'state/configureStore';
import { datasetUpdated } from 'state/actions/actions_projectsmanager';
import errorMessageUtil from 'utils/ErrorMessageUtil';
import getUrl, { pathKey } from 'utils/apiUrls';
import * as apiUtils from 'utils/apiUtils';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { getLanguage } from 'state/constants/getLanguage';
import * as projectsActions from 'state/actions/actions_projects';

const languageConstants = getLanguage();
const { DISPLAY_MESSAGES } = languageConstants;
import * as appActions from './actions_app';
import * as types from './types';

// DATASET

export const openDataset = dataset => ({
  type: types.DATASET_OPEN,
  dataset,
});

export const closeDataset = () => ({
  type: types.DATASET_CLOSE,
});

export const receiveDatasetFail = err => ({
  type: types.RECEIVE_DATASET_FAIL,
  error: err,
});

export const requestTransformDataset = transformingDatasetId => ({
  type: types.REQUEST_DATASET_TRANSFORM,
  transformingDatasetId,
});

export const receiveTransformDatasetProgress = progressData => ({
  type: types.RECEIVE_DATASET_TRANSFORM_PROGRESS,
  progressData,
});

export const receiveTransformDatasetSuccess = datasetId => ({
  type: types.RECEIVE_DATASET_TRANSFORM_SUCCESS,
  value: datasetId,
});

export const receiveTransformDatasetFail = () => ({
  type: types.RECEIVE_DATASET_TRANSFORM_FAIL,
});

export const resetTransformDataset = datasetId => ({
  type: types.RESET_DATASET_TRANSFORM,
  value: datasetId,
});

export const requestDataset = (datasetId, datasetSelectionIndex) => ({
  type: types.REQUEST_DATASET,
  datasetId,
  datasetSelectionIndex,
});

export const receiveDataset = dataset => ({
  type: types.RECEIVE_DATASET,
  dataset,
  receivedAt: Date.now(),
});
// DATASETS BY PROJECT ID

export const requestDatasetsByProjectId = projectId => ({
  type: types.REQUEST_DATASETS_BY_PROJECT_ID,
  requestProjectId: projectId,
});

export const receiveDatasetsByProjectId = datasets => ({
  type: types.RECEIVE_DATASETS_BY_PROJECT_ID,
  datasets,
  receivedAt: Date.now(),
});

// DATASET LIST

export const requestDatasetList = () => ({
  type: types.REQUEST_DATASET_LIST,
});

export const receiveDatasetList = datasetList => ({
  type: types.RECEIVE_DATASET_LIST,
  datasetList,
  receivedAt: Date.now(),
});

// DATASET

export const addDataSetName = name => ({
  type: types.DATASET_ADD_NAME,
  value: name,
});

export const addDataSetDescription = description => ({
  type: types.DATASET_ADD_DESCRIPTION,
  value: description,
});

export const addDataSetType = type => ({
  type: types.DATASET_ADD_TYPE,
  value: type,
});

export const projectDatasetAdd = ({ projectId, datasetId, dataset }) => ({
  type: types.PROJECT_DATASET_ADD,
  projectId,
  datasetId,
  dataset,
});

const datasetRemovedFromProject = ({ projectId, datasetId }) => ({
  type: types.DATASET_REMOVED_FROM_PROJECT,
  projectId,
  datasetId,
});

export const fetchDataset = (datasetId, selectionIndex) => (dispatch, getState) => {
  dispatch(requestDataset(datasetId, selectionIndex));
  const state = getState();
  const { userId, csrfToken } = state.app;
  const clientId = state.header.client.id;
  const projectId = state.projectListSidebar.selectedProjectId;
  let fetchUrl = getUrl(pathKey.datasetById, { datasetId, clientId, projectId });
  return fetch(fetchUrl, {
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      const dataset = apiUtils.normalizeIds(json);
      dispatch(receiveDataset(dataset));
    });
};

export const fetchDatasetTransform = (userId, datasetId, clientId, projectId, csrfToken, isRetry, suggestIntent = false) => {
  const queryParams = queryString.stringify({ useModelForSuggestedCategory: suggestIntent });
  let fetchUrl = getUrl(pathKey.datasetTransform, {
    clientId, projectId, datasetId, queryParams,
  });

  if (isRetry) {
    fetchUrl = getUrl(pathKey.datasetTransformRetry, {
      projectId, datasetId, queryParams, clientId,
    });
  }

  return (dispatch) => {
    dispatch(requestTransformDataset(datasetId));
    return fetch(fetchUrl, {
      credentials: 'same-origin',
      method: 'put',
      headers: {

        'X-CSRF-TOKEN': csrfToken,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ projectId }),
    })
      .then(errorMessageUtil.handleErrors)
      .then(response => response.json())
      .then((json) => {
        dispatch(datasetUpdated({
          projectId,
          datasetId,
          dataset: {
            status: json.status ? (`${json.status}`).toUpperCase() : 'NULL',
            task: json.task ? (`${json.task}`).toUpperCase() : 'NULL',
          },
        }));
      })
      .catch((error) => {
        dispatch(appActions.modalDialogChange(null));
        errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
      });
  };
};

export const fetchDeleteTransformJob = (userId, datasetId, projectId, csrfToken, clientId) => {
  let fetchUrl = getUrl(pathKey.datasetTransform, {
    projectId, datasetId, queryParams: '', clientId,
  });

  return dispatch => fetch(fetchUrl, {
    credentials: 'same-origin',
    method: 'delete',
    headers: {

      'X-CSRF-TOKEN': csrfToken,
      'Content-Type': 'application/json',
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(() => {
      dispatch(resetTransformDataset(datasetId));
      dispatch(datasetUpdated({
        projectId,
        datasetId,
        dataset: { status: 'NULL' },
      }));
    })
    .catch((error) => {
      errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
    });
};

export const fetchCancelTransformJob = (userId, datasetId, projectId, csrfToken, clientId) => {
  let fetchUrl = getUrl(pathKey.datasetTransformCancel, { projectId, datasetId, clientId });

  return dispatch => fetch(fetchUrl, {
    credentials: 'same-origin',
    method: 'put',
    headers: {

      'X-CSRF-TOKEN': csrfToken,
      'Content-Type': 'application/json',
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(resp => resp.json())
    .then((statusResponse) => {
      dispatch(appActions.displayPersistantMessage('Waiting for Transformation Job to be cancelled.'));
      dispatch(receiveTransformDatasetProgress(statusResponse));
      dispatch(datasetUpdated({
        projectId,
        datasetId,
        dataset: { status: 'CANCELLED' },
      }));
    })
    .catch((error) => {
      errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
    });
};

export const addDatasetToProject = ({
  projectId, datasetId, dataset, autoTag,
}) => (dispatch, getState) => new Promise((resolve, reject) => {
  const state = getState();
  const app = state.app;
  const userId = app.userId;
  const csrfToken = app.csrfToken;
  const clientId = state.header.client.id;

  let fetchUrl = getUrl(pathKey.dataset, { projectId, datasetId, clientId });
  if (autoTag) {
    fetchUrl = getUrl(pathKey.addDatasetAutoTag, { projectId, datasetId, clientId });
  }

  fetch(fetchUrl, {
    credentials: 'same-origin',
    method: 'PUT',
    headers: {
      'X-CSRF-TOKEN': csrfToken,

    },
  })
    .then(errorMessageUtil.handleErrors)
    // .then(response => response.json())
    // .then((json) => {
    .then(() => {
      const normalizedDataset = apiUtils.normalizeIds(dataset);
      const newDataset = Object.assign({ _key: `d-${projectId}-${datasetId}`, projectId }, normalizedDataset);
      dispatch(projectDatasetAdd({ projectId, datasetId, dataset: newDataset }));
      resolve();
    })
    .catch((error) => {
      errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
      reject(error);
    });
});

export const removeDatasetFromProject = ({ projectId, datasetId, clientId }) => (dispatch, getState) => new Promise((resolve, reject) => {
  const state = getState(),
    { csrfToken } = state.app;


  let fetchUrl = getUrl(pathKey.dataset, { projectId, datasetId, clientId });
  fetch(fetchUrl, {
    credentials: 'same-origin',
    method: 'delete',
    headers: {
      'X-CSRF-TOKEN': csrfToken,

    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(() => {
      dispatch(projectsActions.refreshProjectsByClient(clientId));
      dispatch(datasetRemovedFromProject({ projectId, datasetId }));
      dispatch(appActions.displayGoodRequestMessage(DISPLAY_MESSAGES.datasetDeleted));
      logAmplitudeEvent(AmplitudeConstants.DELETE_DATASET_COMPLETED_EVENT, state, {
        projectId, datasetId, clientDBId: clientId,
      });
      resolve();
    })
    .catch((error) => {
      logAmplitudeEvent(AmplitudeConstants.DELETE_DATASET_FAILED_EVENT, state, {
        projectId, datasetId, clientDBId: clientId,
      });
      errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
      reject(error);
    });
});
