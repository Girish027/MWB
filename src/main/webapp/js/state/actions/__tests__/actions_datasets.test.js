import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as datasetActions from 'state/actions/actions_datasets';

const middlewares = [thunk];
const reduxMockStore = configureMockStore(middlewares);


describe('actions_datasets', () => {
  let store;
  let props = {
    projectId: '123',
    clientId: '14',
    datasetId: '121',
    dataset: {
      id: '121',
      name: 'dataset',
    },
  };
  const userId = 'testUser';
  const csrfToken = 'jskjfk22';
  const {
    projectId, clientId, datasetId, dataset,
  } = props;

  beforeAll(() => {
    store = reduxMockStore({
      app: {
        userId: 'testUser',
        csrfToken: 'jskjfk22',
      },
      header: {
        client: {
          id: '14',
        },
      },
      projectListSidebar: {
        projectById: {
          123: {
            name: 'OldName',
          },
        },
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
    store.clearActions();
  });

  describe('removeDatasetFromProject', () => {
    test('should dispatch removeDatasetFromProject action with data', () => {
      store.dispatch(datasetActions.removeDatasetFromProject({ projectId, clientId, datasetId }));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('addDatasetToProject', () => {
    test('should dispatch addDatasetToProject action', () => {
      store.dispatch(datasetActions.addDatasetToProject({
        projectId, datasetId, dataset, autoTag: true,
      }));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('fetchCancelTransformJob', () => {
    test('should dispatch fetchCancelTransformJob action', () => {
      store.dispatch(datasetActions.fetchCancelTransformJob(userId, datasetId, projectId, csrfToken, clientId));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('fetchDeleteTransformJob', () => {
    test('should dispatch fetchDeleteTransformJob action', () => {
      store.dispatch(datasetActions.fetchDeleteTransformJob(userId, datasetId, projectId, csrfToken, clientId));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('fetchDatasetTransform', () => {
    test('should dispatch fetchDatasetTransform action', () => {
      store.dispatch(datasetActions.fetchDatasetTransform(userId, datasetId, clientId, projectId, csrfToken, true));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('fetchDataset', () => {
    test('should dispatch fetchDataset action', () => {
      store.dispatch(datasetActions.fetchDataset(datasetId, 0));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
