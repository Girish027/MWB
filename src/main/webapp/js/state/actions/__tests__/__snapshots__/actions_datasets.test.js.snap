// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions_datasets addDatasetToProject should dispatch addDatasetToProject action 1`] = `Array []`;

exports[`actions_datasets fetchCancelTransformJob should dispatch fetchCancelTransformJob action 1`] = `Array []`;

exports[`actions_datasets fetchDataset should dispatch fetchDataset action 1`] = `
Array [
  Object {
    "datasetId": "121",
    "datasetSelectionIndex": 0,
    "type": "REQUEST_DATASET",
  },
]
`;

exports[`actions_datasets fetchDatasetTransform should dispatch fetchDatasetTransform action 1`] = `
Array [
  Object {
    "transformingDatasetId": "121",
    "type": "REQUEST_DATASET_TRANSFORM",
  },
]
`;

exports[`actions_datasets fetchDeleteTransformJob should dispatch fetchDeleteTransformJob action 1`] = `Array []`;

exports[`actions_datasets removeDatasetFromProject should dispatch removeDatasetFromProject action with data 1`] = `Array []`;
